state/protocol/validity.go

```diff
@@ -14,20 +14,20 @@ import (
 // CAUTION: This function assumes that all inputs besides extendingCommit are already validated.
 // Expected errors during normal operations:
 // * protocol.InvalidServiceEventError if the input service event is invalid to extend the currently active epoch status
-func IsValidExtendingEpochSetup(extendingSetup *flow.EpochSetup, protocolStateEntry *flow.ProtocolStateEntry, currentEpochSetupEvent *flow.EpochSetup) error {
+func IsValidExtendingEpochSetup(extendingSetup *flow.EpochSetup, protocolStateEntry *flow.RichProtocolStateEntry) error {
```

Comment:
I think changing the `protocolStateEntry`'s type to `RichProtocolStateEntry` strongly increases the risk of accidental bugs. This is because we do not maintain a `RichProtocolStateEntry` while evolving the state. Therefore, we create the incentive to use the parent state, which is unaware of the evolution the protocol state already has undergone. 

I could imagine that the resulting code would accept multiple `EpochSetup` events as they are all consistent with the parent state. Though, repeated `EpochSetup` events should technically be rejected. 

I would suggest to address this as part of a later PR. Created issue https://github.com/onflow/flow-go/issues/6019 which also describes the problem in more detail